astropy/units/tests/test_quantity_decorator.py

```diff
@@ -190,3 +190,63 @@ def myfunc_args(solarx, **kwargs):
     assert isinstance(solarx, u.Quantity)
 
     assert solarx.unit == u.deg
+
+# ---------------------------------------------------------------
+# Repeat core tests with physical types instead of units
```

Comment:
Might be better to just use `pytest.mark.parametrize`; that way the tests cannot get out of sync.